chain/boltdb/store.go

```diff
@@ -52,6 +73,35 @@ func NewBoltStore(l log.Logger, folder string, opts *bolt.Options) (*BoltStore,
 	}, err
 }
 
+func shouldUseTrimmedBolt(ctx context.Context, sourceBeaconPath string, opts *bolt.Options) bool {
+	if isThisATest(ctx) {
+		return false
+	}
+
+	// New beacons stores should use the trimmed version
+	if _, err := os.Stat(sourceBeaconPath); errors.Is(err, os.ErrNotExist) {
+		return true
+	}
+
+	// Existing beacon stores should use the format that's suitable
+	existingDB, err := bolt.Open(sourceBeaconPath, BoltStoreOpenPerm, opts)
+	if err != nil {
+		return true
+	}
+	defer func() {
+		_ = existingDB.Close()
+	}()
+
+	err = existingDB.View(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(beaconBucket)
+		_, value := bucket.Cursor().First()
+		b := chain.Beacon{}
+		return json.Unmarshal(value, &b)
+	})
```

Comment:
It will result in an error because `json.Unmarshal` will not be able to extract a JSON from the stored value. In that case, we assume we should use the new format. If that encounters  another error, then we'll report it on that execution path.

There is a test for this in `TestShouldUseTrimmedBolt`.

Reply:
I don't think a beacon using the trimmed format would error out on Unmarshalling, it would just have a "null" previous_signature field.

Reply:
it would error out because now we're not storing json in boltdb anymore. Resolving this.